mempool/cat/pool.go

```diff
@@ -525,7 +567,20 @@ func (txmp *TxPool) Update(
 	txmp.metrics.SuccessfulTxs.Add(float64(len(blockTxs)))
 	for _, tx := range blockTxs {
 		// Regardless of success, remove the transaction from the mempool.
-		txmp.removeTxByKey(tx.Key())
+		txKey := tx.Key()
+		wtx := txmp.store.get(txKey)
+		txmp.removeTxByKey(txKey)
+		// Write trace for confirmed transaction if we have the wrapped tx info
+		if wtx != nil {
+			schema.WriteMempoolTxStatus(
+				txmp.traceClient,
+				txKey[:],
+				schema.TxConfirmed,
+				nil,
+				wtx.sender,
+				wtx.sequence,
+			)
+		}
```

Comment:
the tx status stuff, particularly this one is actually a nice tracing option so happy to remove this in a different PR or just keep